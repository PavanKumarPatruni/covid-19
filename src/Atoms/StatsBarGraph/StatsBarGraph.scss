@import '../../styles/index.scss';

.graph-title {
  left: 54px;
  top: 16px;
}
