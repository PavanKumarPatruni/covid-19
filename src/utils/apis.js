const rootAPI = 'https://api.covid19india.org';

export default {
  dataAPI: `${rootAPI}/data.json`,
  stateDistrictWiseAPI: `${rootAPI}/state_district_wise.json`,
};
