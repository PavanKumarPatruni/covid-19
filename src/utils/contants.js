export const Colors = {
  GREEN: '#28a745',
  RED: '#ff073a',
  BLUE: '#007bff',
  GREY: '#827397',
  WHITE: '#FFFFFF',
  PRIMARY: '#201c3a',
  TRANSPARENT: '#00000000',
};
