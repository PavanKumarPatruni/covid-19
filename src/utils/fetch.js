/* eslint-disable compat/compat */
export const fetchGetData = apiData => {
  return fetch(apiData.url, {
    method: 'GET',
  }).then(res => res.json());
};
