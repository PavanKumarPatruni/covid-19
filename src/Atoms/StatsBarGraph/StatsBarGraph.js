import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { Line } from 'react-chartjs-2';

import { Colors } from '../../utils/contants';

import './StatsBarGraph.scss';

const propTypes = {
  title: PropTypes.string,
  type: PropTypes.string,
  stats: PropTypes.array,
  color: PropTypes.string,
};

const StatsBarGraph = ({ title, type, stats, color }) => {
  const [statsData, setStatsData] = useState({});

  useEffect(() => {
    if (stats && stats.length > 0) {
      const labels = stats.map(() => '');
      const array = stats.map(item => parseInt(item[type], 10));

      const data = {
        labels,
        datasets: [
          {
            fill: false,
            lineTension: 0.1,
            backgroundColor: Colors.TRANSPARENT,
            borderColor: color,
            borderCapStyle: 'butt',
            borderDash: [],
            borderDashOffset: 0.0,
            borderJoinStyle: 'miter',
            pointBorderColor: color,
            pointBackgroundColor: color,
            pointBorderWidth: 1,
            pointHoverRadius: 5,
            pointHoverBackgroundColor: color,
            pointHoverBorderColor: Colors.WHITE,
            pointHoverBorderWidth: 1,
            pointRadius: 1,
            pointHitRadius: 10,
            data: array,
          },
        ],
      };

      setStatsData(data);
    }
  }, [stats]);

  return (
    <div className="relative flex flex-justify-center mt4">
      <h3 className="absolute graph-title" style={{ color }}>
        {title}
      </h3>
      <Line
        width={400}
        height={250}
        data={statsData}
        legend={{ display: false }}
      />
    </div>
  );
};

StatsBarGraph.defaultProps = {
  title: '',
  type: '',
  stats: [],
  color: '',
};

StatsBarGraph.propTypes = propTypes;

export default StatsBarGraph;
