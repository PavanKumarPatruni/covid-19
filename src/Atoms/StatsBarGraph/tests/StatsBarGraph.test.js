import React from 'react';
import { shallow, mount } from 'enzyme';

import StatsBarGraph from '../StatsBarGraph';

import { Colors } from '../../../utils/contants';

describe('StatsBarGraph component', () => {
  const props = {
    title: 'Confirmed',
    type: 'confirmed',
    stats: [
      {
        active: '44351',
        confirmed: '67700',
        deaths: '2215',
        deltaconfirmed: '524',
        deltadeaths: '2',
        deltarecovered: '160',
        lastupdatedtime: '11/05/2020 15:41:22',
        recovered: '21130',
        state: 'Total',
        statecode: 'TT',
        statenotes: '',
      },
      {
        active: '44351',
        confirmed: '67700',
        deaths: '2215',
        deltaconfirmed: '524',
        deltadeaths: '2',
        deltarecovered: '160',
        lastupdatedtime: '11/05/2020 15:41:22',
        recovered: '21130',
        state: 'Total',
        statecode: 'AP',
        statenotes: '',
      },
      {
        active: '44351',
        confirmed: '67700',
        deaths: '2215',
        deltaconfirmed: '524',
        deltadeaths: '0',
        deltarecovered: '160',
        lastupdatedtime: '11/05/2020 15:41:22',
        recovered: '21130',
        state: 'Total',
        statecode: 'TN',
        statenotes: '',
      },
    ],
    color: Colors.RED,
  };

  test('match snapshot correctly', () => {
    const wrapper = shallow(<StatsBarGraph {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('match snapshot correctly - with empty array', () => {
    props.stats = [];
    const wrapper = shallow(<StatsBarGraph {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('match snapshot correctly - with empty title', () => {
    props.title = '';
    const wrapper = shallow(<StatsBarGraph {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('match snapshot correctly - with empty title', () => {
    props.title = '';
    const wrapper = mount(<StatsBarGraph {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('match snapshot correctly- with empty color', () => {
    props.color = '';
    const wrapper = shallow(<StatsBarGraph {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('match snapshot correctly - with empty color', () => {
    props.color = '';
    const wrapper = mount(<StatsBarGraph {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('match snapshot correctly- with empty type', () => {
    props.type = '';
    const wrapper = shallow(<StatsBarGraph {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('match snapshot correctly - with empty type', () => {
    props.type = '';
    const wrapper = mount(<StatsBarGraph {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
