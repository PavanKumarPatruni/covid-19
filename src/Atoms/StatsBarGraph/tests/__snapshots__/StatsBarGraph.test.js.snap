// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsBarGraph component match snapshot correctly - with empty array 1`] = `
<div
  className="relative flex flex-justify-center mt4"
>
  <h3
    className="absolute graph-title"
    style={
      Object {
        "color": "#ff073a",
      }
    }
  >
    Confirmed
  </h3>
  <Line
    data={Object {}}
    height={250}
    legend={
      Object {
        "display": false,
      }
    }
    width={400}
  />
</div>
`;

exports[`StatsBarGraph component match snapshot correctly - with empty color 1`] = `
<StatsBarGraph
  color=""
  stats={Array []}
  title=""
  type="confirmed"
>
  <div
    className="relative flex flex-justify-center mt4"
  >
    <h3
      className="absolute graph-title"
      style={
        Object {
          "color": "",
        }
      }
    />
    <Line
      data={
        Object {
          "datasets": Array [],
          "labels": Array [],
        }
      }
      height={250}
      legend={
        Object {
          "display": false,
        }
      }
      width={400}
    >
      <ChartComponent
        data={
          Object {
            "datasets": Array [],
            "labels": Array [],
          }
        }
        datasetKeyProvider={[Function]}
        height={250}
        legend={
          Object {
            "display": false,
          }
        }
        options={
          Object {
            "legend": Object {
              "display": false,
            },
          }
        }
        redraw={false}
        type="line"
        width={400}
      >
        <canvas
          height={250}
          onClick={[Function]}
          width={400}
        />
      </ChartComponent>
    </Line>
  </div>
</StatsBarGraph>
`;

exports[`StatsBarGraph component match snapshot correctly - with empty title 1`] = `
<div
  className="relative flex flex-justify-center mt4"
>
  <h3
    className="absolute graph-title"
    style={
      Object {
        "color": "#ff073a",
      }
    }
  />
  <Line
    data={Object {}}
    height={250}
    legend={
      Object {
        "display": false,
      }
    }
    width={400}
  />
</div>
`;

exports[`StatsBarGraph component match snapshot correctly - with empty title 2`] = `
<StatsBarGraph
  color="#ff073a"
  stats={Array []}
  title=""
  type="confirmed"
>
  <div
    className="relative flex flex-justify-center mt4"
  >
    <h3
      className="absolute graph-title"
      style={
        Object {
          "color": "#ff073a",
        }
      }
    />
    <Line
      data={
        Object {
          "datasets": Array [],
          "labels": Array [],
        }
      }
      height={250}
      legend={
        Object {
          "display": false,
        }
      }
      width={400}
    >
      <ChartComponent
        data={
          Object {
            "datasets": Array [],
            "labels": Array [],
          }
        }
        datasetKeyProvider={[Function]}
        height={250}
        legend={
          Object {
            "display": false,
          }
        }
        options={
          Object {
            "legend": Object {
              "display": false,
            },
          }
        }
        redraw={false}
        type="line"
        width={400}
      >
        <canvas
          height={250}
          onClick={[Function]}
          width={400}
        />
      </ChartComponent>
    </Line>
  </div>
</StatsBarGraph>
`;

exports[`StatsBarGraph component match snapshot correctly - with empty type 1`] = `
<StatsBarGraph
  color=""
  stats={Array []}
  title=""
  type=""
>
  <div
    className="relative flex flex-justify-center mt4"
  >
    <h3
      className="absolute graph-title"
      style={
        Object {
          "color": "",
        }
      }
    />
    <Line
      data={
        Object {
          "datasets": Array [],
          "labels": Array [],
        }
      }
      height={250}
      legend={
        Object {
          "display": false,
        }
      }
      width={400}
    >
      <ChartComponent
        data={
          Object {
            "datasets": Array [],
            "labels": Array [],
          }
        }
        datasetKeyProvider={[Function]}
        height={250}
        legend={
          Object {
            "display": false,
          }
        }
        options={
          Object {
            "legend": Object {
              "display": false,
            },
          }
        }
        redraw={false}
        type="line"
        width={400}
      >
        <canvas
          height={250}
          onClick={[Function]}
          width={400}
        />
      </ChartComponent>
    </Line>
  </div>
</StatsBarGraph>
`;

exports[`StatsBarGraph component match snapshot correctly 1`] = `
<div
  className="relative flex flex-justify-center mt4"
>
  <h3
    className="absolute graph-title"
    style={
      Object {
        "color": "#ff073a",
      }
    }
  >
    Confirmed
  </h3>
  <Line
    data={Object {}}
    height={250}
    legend={
      Object {
        "display": false,
      }
    }
    width={400}
  />
</div>
`;

exports[`StatsBarGraph component match snapshot correctly- with empty color 1`] = `
<div
  className="relative flex flex-justify-center mt4"
>
  <h3
    className="absolute graph-title"
    style={
      Object {
        "color": "",
      }
    }
  />
  <Line
    data={Object {}}
    height={250}
    legend={
      Object {
        "display": false,
      }
    }
    width={400}
  />
</div>
`;

exports[`StatsBarGraph component match snapshot correctly- with empty type 1`] = `
<div
  className="relative flex flex-justify-center mt4"
>
  <h3
    className="absolute graph-title"
    style={
      Object {
        "color": "",
      }
    }
  />
  <Line
    data={Object {}}
    height={250}
    legend={
      Object {
        "display": false,
      }
    }
    width={400}
  />
</div>
`;
